src/pages/Bookmarks.jsx

```diff
@@ -2,15 +2,21 @@ import React, { useContext, useState, useEffect } from 'react'
 import Header from '../components/Navbar'
 import { ToolContext } from '../App'
 import noresultimg from '../assets/sad-face.png'
-import { useSearchParams } from 'react-router-dom'
+import { useSearchParams, Navigate } from 'react-router-dom'
 import filterProducts from '../utils/filter/filter_products'
+import { allFilters } from '../utils/data/filters'
 
 const BookMarks = ({ length }) => {
   const { bookmarkfilteredProducts, deleteres } =
     useContext(ToolContext);
   const [searchParams] = useSearchParams()
   let filters = searchParams.getAll('filters').length > 0 ? searchParams.getAll('filters') : ["all"]
```

Comment:
what's the use of getAll if you are using the filter's first index to search
`filters = filters[0].split(",")`